Extend error reporting when creating note.

c-okelly opened the pull request:
In reference to #72.

If the user adds an invalid note to Anki anki-connect will return a more specific error message.

Added a string method for class AnkiNoteParams.

Let me know if you want anything changed.

C

FooSoft commented on AnkiConnect.py:
```diff
@@ -89,11 +89,11 @@ def makeStr(data):
-    except web.URLError:
-        return None
+    except web.URLError as e:
+        raise Exception("A urlError has occoured for url " + url + ". Error messages was: " + e.message)
```
Single quotes instead of double quotes please (all throughout this PR)

c-okelly replied:
Will do.

FooSoft commented on AnkiConnect.py:
```diff
@@ -89,11 +89,11 @@ def makeStr(data):
 
     if resp.code != 200:
-        return None
+        raise Exception("Return conde for url request" + url + "was not 200. Error code: " + resp.code)
```
conde => code

c-okelly replied:
Corrected

FooSoft commented on AnkiConnect.py:
```diff
@@ -1110,11 +1118,14 @@ def addNote(self, note):
+                else:
+                    results.append(None)
+            except Exception as e:
+                results.append(str(e))
```
This change would mean that `addNotes` can return a note ID, None or a string. Existing code in things that use AnkiConnect likely have code that checks to see if result `is not None`, and this would break their behavior.

c-okelly replied:
Changed to append(None) instead

c-okelly commented:
Will change it to return just none.

c-okelly commented:
I defiantly agree that line 1127 by introducing string return could break functionality. I have changed it so that the when an exception occurs it just append an None to the array.

The code I have added done on line 428-435 does mean now that if the card was a duplicate or an empty card it will throw an error. I would be concerned this might be more hassle then a benefit if people use the api to add a lot of cards in one go if the exception is not caught but it does also mean the information is buried. 

If you think it should be removed will do or if there are any other changes you think could be me let me know.

Thanks

FooSoft commented:
Thanks for the changes. I see your point about throwing exceptions half way through an operation, but I don't really think there is any other good way to report multiple errors through the API.

c-okelly commented:
No neither do it as it's is currently tbh. Thanks for the feedback though.
